core/src/main/java/io/parsingdata/metal/expression/value/arithmetic/Mul.java

```diff
@@ -35,7 +35,7 @@ public Mul(final ValueExpression left, final ValueExpression right) {
     }
 
     @Override
-    public Optional<Value> eval(final Value left, final Value right, final Environment environment, final Encoding encoding) {
+    public Optional<Value> eval(final Value left, final Value right, final ParseGraph graph, final Encoding encoding) {
         return Optional.of(ConstantFactory.createFromNumeric(left.asNumeric().multiply(right.asNumeric()), encoding));
     }
```

Comment:
Correct! I would like to discuss this a bit further. There is some low-hanging fruit, but if we go this path we need to go all-in and I want to make sure we understand what that entails.